packages/grafana-ui/src/utils/fieldDisplay.ts

```diff
@@ -0,0 +1,278 @@
+import toNumber from 'lodash/toNumber';
+import toString from 'lodash/toString';
+
+import {
+  ValueMapping,
+  Threshold,
+  DisplayValue,
+  FieldType,
+  NullValueMode,
+  GrafanaTheme,
+  SeriesData,
+  InterpolateFunction,
+  Field,
+  ScopedVars,
+  GraphSeriesValue,
+} from '../types/index';
+import { getDisplayProcessor } from './displayValue';
+import { getFlotPairs } from './flotPairs';
+import { ReducerID, reduceField } from './fieldReducer';
+
+export interface FieldDisplayOptions {
+  values?: boolean; // If true show each row value
+  limit?: number; // if showing all values limit
+  calcs: string[]; // when !values, pick one value for the whole field
+
+  defaults: Partial<Field>; // Use these values unless otherwise stated
+  override: Partial<Field>; // Set these values regardless of the source
+
+  // Could these be data driven also?
+  thresholds: Threshold[];
+  mappings: ValueMapping[];
+}
+
+export const VAR_SERIES_NAME = '__series_name';
+export const VAR_FIELD_NAME = '__field_name';
+export const VAR_CALC = '__calc';
+export const VAR_CELL_PREFIX = '__cell_'; // consistent with existing table templates
```

Comment:
Naming for this one feel unintuitive, but not sure what is better.

Reply:
maybe __field_N?  or or _field_N_value?

Reply:
Yea had the same thought, tricky since it easily confused with field name. __field_N is think the best one